Adapt document classes

kdeininger opened the pull request:
Closes #1176 and #1249

vvmruder reviewed (changes requested):
There are 18 tests failing for now when I do a `make clean test`. Especially the tests about `FAILED tests/renderer/test_json.py::test_format_` should be fixed.

```
=========================================== short test summary info ============================================
FAILED tests/renderer/test_json.py::test_format_real_estate - TypeError: 'NoneType' object is not subscriptable
FAILED tests/renderer/test_json.py::test_format_plr[parameter0] - TypeError: 'NoneType' object is not subscri...
FAILED tests/renderer/test_json.py::test_format_plr[parameter1] - TypeError: 'NoneType' object is not subscri...
FAILED tests/renderer/test_json.py::test_format_document[document0-result_dict0] - TypeError: 'NoneType' obje...
FAILED tests/renderer/test_json.py::test_format_document[document1-result_dict1] - TypeError: 'NoneType' obje...
FAILED tests/renderer/test_json.py::test_format_document[document2-result_dict2] - TypeError: 'NoneType' obje...
FAILED tests/renderer/test_xml.py::test_extract_against_schema[parameter0-test_extract0] - lxml.etree.Documen...
FAILED tests/renderer/test_xml.py::test_extract_against_schema[parameter1-test_extract1] - lxml.etree.Documen...
FAILED tests/renderer/test_xml.py::test_extract_against_schema[parameter2-test_extract2] - lxml.etree.Documen...
FAILED tests/webservice/test_case_insensitivity.py::test_get_extract_by_id[params0] - TypeError: 'NoneType' o...
FAILED tests/webservice/test_case_insensitivity.py::test_get_extract_by_id[params1] - TypeError: 'NoneType' o...
FAILED tests/webservice/test_case_insensitivity.py::test_get_extract_by_id[params2] - TypeError: 'NoneType' o...
FAILED tests/webservice/test_case_insensitivity.py::test_get_extract_by_id[params3] - TypeError: 'NoneType' o...
FAILED tests/webservice/test_case_insensitivity.py::test_get_extract_by_id[params4] - TypeError: 'NoneType' o...
FAILED tests/webservice/test_case_insensitivity.py::test_get_extract_by_id[params5] - TypeError: 'NoneType' o...
FAILED tests/webservice/test_getextractbyid.py::test_return_json[ALL] - TypeError: 'NoneType' object is not s...
FAILED tests/webservice/test_getextractbyid.py::test_return_json[ALL_FEDERAL] - TypeError: 'NoneType' object ...
FAILED tests/xtf_import/test_federal_topic.py::test_load - SystemExit: 1
============================ 18 failed, 356 passed, 1 warning in 181.94s (0:03:01) =============================
make: *** [Makefile:151: test] Error 1
```

Otherwise it looks good.

kdeininger commented:
I have rebased this branch onto master and fixed some tests for the `OEREBlexSource` that occurred after switching the GEO-Link schema to version 1.2.1. Now I have again only the four known tests failing.

@vvmruder, did you rebuild your local devcontainers and run a `make clean-all` before running the checks?

vuilleumierc commented:
I also have only 4 failing tests.
But there are some issues with `make serve-dev`:
1. I have to run it twice for it to complete, on the first try I get a Python error from load_sample_data.py
2. The service does not seem to work, the URL `http://localhost:6543/oereb/extract/reduced/xml/CH113928077734` gives an empty content (same with JSON). But maybe this is normal at this point?

vvmruder commented:
I tried it now and the tests are ok.

But now I get an error while dev date is filled into the db:
```
.venv/bin/python pyramid_oereb/standard/load_sample_data.py --configuration pyramid_oereb/standard/pyramid_oereb.yml --sql-file .db/13-fill.sql --dir sample_data
Import theme contaminated_public_transport_sites.
Traceback (most recent call last):
  File "pyramid_oereb/standard/load_sample_data.py", line 329, in <module>
    _run()
  File "pyramid_oereb/standard/load_sample_data.py", line 324, in _run
    sql_file=sql_file
  File "pyramid_oereb/standard/load_sample_data.py", line 236, in load
    ][0]
IndexError: list index out of range
make: *** [Makefile:106: .db/13-fill.sql] Error 1
```

for `make serve-dev`

kdeininger commented:
Okay, I will fix the sample data.

kdeininger commented:
The sample data import should work now.

vuilleumierc commented:
Looks good to me, thanks!
I have one comment in the code

vuilleumierc commented on pyramid_oereb/lib/records/documents.py:
```diff
@@ -6,317 +6,115 @@
         if published_from and not isinstance(published_from, datetime.date):
             warnings.warn('Type of "published_from" should be "datetime.date", not '
                           + str(type(published_from)))
 
```
shouldn't we test here for the type of the new attributes `only_in_municipality`, `article_nb` and `published_until`?

vvmruder replied:
This might be a good completion :D

kdeininger replied:
Done 😉

vvmruder commented:
Maybe solve the comment above. Otherwise Iam happy with it and thank you for this bunch of work.
